src/Form/TfaLoginForm.php

```diff
@@ -132,11 +134,85 @@ public function submitForm(array &$form, FormStateInterface $form_state) {
    *
    * @param FormStateInterface $form_state
    */
-  function TfaLoginFormRedirect($form, &$form_state){
+  public function TfaLoginFormRedirect($form, &$form_state){
     $route = $form_state->getValue('tfa_redirect');
     if (isset($route)) {
       $form_state->setRedirect($route);
     }
   }
 
+  /**
+   * Check if TFA process has completed so authentication should not be stopped.
+   *
+   * @param $account User account
+   * @return bool
+   */
+  protected function loginComplete($account) {
+    // TFA master login allowed switch is set by tfa_login().
+//    $tfa_session = $this->session->get('tfa');
+//    if (isset($tfa_session[$account->id()]['login']) && $tfa_session[$account->id()]['login'] === TRUE) {
+//      return TRUE;
+//    }
+    return FALSE;
+  }
+
+
+  /**
+   * Determine if TFA process is ready.
+   *
+   * @param \Drupal\tfa\Plugin\TfaValidationInterface $tfaValidationPlugin
+   * @return bool Whether process can begin or not.
+   */
+  protected function ready(TfaValidationInterface $tfaValidationPlugin) {
+    return $tfaValidationPlugin->ready();
+  }
+
+  /**
+   * Whether authentication should be allowed and not interrupted.
+   *
+   * If any plugin returns TRUE then authentication is not interrupted by TFA.
+   *
+   * @param \Drupal\tfa\Plugin\TfaLoginInterface $tfaLogin
+   * @return bool
+   */
+  protected function loginAllowed(TfaLoginInterface $tfaLoginPlugin) {
+//    if (!empty($tfaLoginPlugin)) {
+//      foreach ($tfaLoginPlugin as $class) {
+//        if ($class->loginAllowed()) {
+//          return TRUE;
+//        }
+//      }
+//    }
+    return FALSE;
+  }
+
+  /**
+   * Begin the TFA process.
+   */
+  protected function begin(TfaValidationInterface $tfaValidationPlugin) {
+    // Invoke begin method on send validation plugins.
+    if (method_exists($tfaValidationPlugin, 'begin')) {
+      $tfaValidationPlugin->begin();
+    }
+  }
+
+
+  /**
+   * Generate account hash to access the TFA form.
+   *
+   * @param object $account User account.
+   * @return string Random hash.
+   */
+  //function tfa_login_hash($account) {
+  protected function getLoginHash($account) {
```

Comment:
Going back and forth between needing a service and not needing a service. I think it will end up with the former, where we do have a service to handle some of this. For now, this was the only place it was being used (other than commented code) so just cleaning it up. Once there is a 2nd location that needs it, I will move it to a service.